app/pages/sections/additional-services/order-item/controller.js

```diff
@@ -0,0 +1,187 @@
+import { logger } from '../../../../logger';
+import { getEndpoint } from '../../../../endpoints';
+import commonManifest from './commonManifest.json';
+import { getSelectedPriceManifest } from '../../../../helpers/controllers/manifestProvider';
+import { extractDate } from '../../../../helpers/controllers/extractDate';
+
+const formatFormData = ({ formData }) => ({
+  'deliveryDate-day': formData['deliveryDate-day']
+    ? formData['deliveryDate-day'].trim() : undefined,
+  'deliveryDate-month': formData['deliveryDate-month']
+    ? formData['deliveryDate-month'].trim() : undefined,
+  'deliveryDate-year': formData['deliveryDate-year']
+    ? formData['deliveryDate-year'].trim() : undefined,
+  quantity: formData.quantity
+    ? formData.quantity.trim() : undefined,
+  price: formData.price && formData.price.length > 0
+    ? formData.price.trim() : undefined,
+  selectEstimationPeriod: formData.selectEstimationPeriod
+    ? formData.selectEstimationPeriod.trim() : undefined,
+});
+
+export const getOrderItemContext = async ({
+  orderId,
+  orderItemId,
+  orderItemType,
+  itemName,
+  odsCode,
+  serviceRecipientName,
+  selectedPrice,
+  formData,
+}) => {
+  const selectedPriceManifest = getSelectedPriceManifest({
+    orderItemType,
+    provisioningType: selectedPrice.provisioningType,
+    type: selectedPrice.type,
+  });
+
+  return getContext({
+    commonManifest,
+    selectedPriceManifest,
+    orderId,
+    orderItemId,
+    itemName,
+    serviceRecipientName,
+    odsCode,
+    selectedPrice,
+    formData,
+  });
+};
+
+export const getOrderItemErrorPageContext = (params) => {
+  const formattedData = formatFormData({
+    formData: params.formData,
+  });
+
+  const selectedPriceManifest = getSelectedPriceManifest({
+    orderItemType: params.orderItemType,
+    provisioningType: params.selectedPrice.provisioningType,
+    type: params.selectedPrice.type,
+  });
+
+  const updatedParams = {
+    ...params,
+    commonManifest,
+    selectedPriceManifest,
+    formData: formattedData,
+  };
+
+  return getErrorContext(updatedParams);
+};
+
+const formatPostData = ({
+  selectedRecipientId,
+  serviceRecipientName,
+  selectedSolutionId,
+  solutionName,
+  selectedPrice,
+  formData,
+}) => ({
+  ...selectedPrice,
+  serviceRecipient: {
+    name: serviceRecipientName,
+    odsCode: selectedRecipientId,
+  },
+  catalogueSolutionId: selectedSolutionId,
+  catalogueSolutionName: solutionName,
+  deliveryDate: extractDate('deliveryDate', formData),
+  quantity: parseInt(formData.quantity, 10),
+  estimationPeriod: formData.selectEstimationPeriod,
+  price: parseFloat(formData.price),
+});
+
+const postSolutionOrderItem = async ({
+  accessToken,
+  orderId,
+  selectedRecipientId,
+  serviceRecipientName,
+  selectedSolutionId,
+  solutionName,
+  selectedPrice,
+  formData,
+}) => {
+  const endpoint = getEndpoint({ api: 'ordapi', endpointLocator: 'postSolutionOrderItem', options: { orderId } });
```

Comment:
Same here? `postSolutionOrderItem` should this be `postAdditionalServiceOrderItem`? Or could this be `postOrderItem`?

Reply:
yep postOrderItem. coming soon 👍